wifi/src/sta.cpp

```diff
@@ -0,0 +1,198 @@
+#include "sta.hpp"
+
+#include <string.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/event_groups.h"
+#include "freertos/task.h"
+#include "esp_mac.h"
+#include "esp_wifi.h"
+#include "esp_system.h"
+#include "esp_event.h"
+#include "esp_log.h"
+#include "nvs_flash.h"
+
+#include "lwip/err.h"
+#include "lwip/sys.h"
+#include "lwip/sockets.h"
+
+#define STA_CONNECTED_BIT		BIT0
+#define STA_DISCONNECTED_BIT	BIT1
+
+namespace sdk
+{
+	namespace wifi {
+
+	sta::sta(sta_config_t config) : m_config(config) {
+        event_group = xEventGroupCreate();
+	}
+
+	res sta::get_status()
+	{
+		return Ok(RUNNING);
+	}
+
+	res sta::run()
+	{
+		return Ok(RUNNING);
+	}
```

Comment:
For all of these overrides, they should probably return some `m_state` variable. So that you can update them accordingly internally.